### PR TITLE
Add tests around finishing quests

### PR DESCRIPTION
**Pull request by crookedneighbor**
Also adds a virtual method to return an array of participating quest members.

### CODE REVIEW

File: website/server/models/group.js [hunk 1]
```diff
@@ -478,7 +485,7 @@ schema.methods.finishQuest = async function finishQuest (quest) {
     }
   });
 
-  let q = this._id === TAVERN_ID ? {} : {_id: {$in: _.keys(this.quest.members)}};
```
Comment (crookedneighbor) [inline website/server/models/group.js:481]: We were awarding quest rewards to all members, not just participating ones :(

File: website/server/models/group.js [hunk 2]
```diff
@@ -439,11 +443,14 @@ schema.statics.cleanGroupQuest = function cleanGroupQuest () {
...
   let questK = quest.key;
-  let updates = {$inc: {}, $set: {}};
-
-  updates.$inc[`achievements.quests.${questK}`] = 1;
-  updates.$inc['stats.gp'] = Number(quest.drop.gp);
-  updates.$inc['stats.exp'] = Number(quest.drop.exp);
+  let updates = {
+    $inc: {
+      [`achievements.quests.${questK}`]: 1,
+      ['stats.gp']: Number(quest.drop.gp),
```
Comment (paglias) [inline website/server/models/group.js]: no need for square brackets here

File: website/server/models/group.js [hunk 3]
```diff
@@ -90,6 +90,10 @@ schema.plugin(baseModel, {
   noSet: ['_id', 'balance', 'quest', 'memberCount', 'chat', 'challengeCount'],
 });
 
+schema.virtual('quest.participatingMembers').get(function getParticipatingQuestMembers () {
```
Comment (paglias) [inline website/server/models/group.js]: can we transform this in a method? I find virtual getters and setters confusing, with a method you immediately now that you're computing a result

File: website/server/models/group.js (hunk 1 above)
Comment (khipkin) [inline website/server/models/group.js:481]: Semi-related to #7653

File: website/server/models/group.js (hunk 2 above)
Comment (crookedneighbor) [inline website/server/models/group.js]: Good catch.

File: website/server/models/group.js (hunk 3 above)
Comment (crookedneighbor) [inline website/server/models/group.js]: I could go either way on this, but I'm fine making it an instance method.

**Pull request merged**